src/mon/LogMonitor.cc

```diff
@@ -699,7 +702,12 @@ bool LogMonitor::preprocess_log(MonOpRequestRef op)
 	    << session->caps << dendl;
     goto done;
   }
-  
+
+  if (last_to_remove == trim_max * trim_max_multiplier) {
```

Comment:
We should check that `trim_max_multiplier` actually has a value. if `trim_max_multiplier` equals to zero trim max is disabled and we would ignore all cases in which `last_to_remove` is zero.
```
if (trim_max_multiplier && last_to_remove == trim_max * trim_max_multiplier)
```
should probably do.

Reply:
done